addOns/exim/src/main/java/org/zaproxy/addon/exim/har/HarUtils.java

```diff
@@ -0,0 +1,447 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2024 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.exim.har;
+
+import de.sstoehr.harreader.model.HarCache;
+import de.sstoehr.harreader.model.HarContent;
+import de.sstoehr.harreader.model.HarCookie;
+import de.sstoehr.harreader.model.HarCreatorBrowser;
+import de.sstoehr.harreader.model.HarEntry;
+import de.sstoehr.harreader.model.HarHeader;
+import de.sstoehr.harreader.model.HarLog;
+import de.sstoehr.harreader.model.HarPostData;
+import de.sstoehr.harreader.model.HarPostDataParam;
+import de.sstoehr.harreader.model.HarQueryParam;
+import de.sstoehr.harreader.model.HarRequest;
+import de.sstoehr.harreader.model.HarResponse;
+import de.sstoehr.harreader.model.HarTiming;
+import de.sstoehr.harreader.model.HttpMethod;
+import de.sstoehr.harreader.model.HttpStatus;
+import java.net.HttpCookie;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.network.HtmlParameter;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpHeaderField;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.zaproxy.zap.network.HttpRequestBody;
+
+/**
+ * Utility class to parse/create HTTP Archives (HAR) and do conversions between HAR Java classes and
+ * {@code HttpMessage}s (request and response).
+ *
+ * @see <a href="http://www.softwareishard.com/blog/har-12-spec/">HTTP Archive 1.2</a>
+ * @see HttpMessage
+ */
+public final class HarUtils {
```

Comment:
This is largely an updated version of what's in core. ~85% of the code is duplicated.

Reply:
Is this one going to replace the core class to export?

Reply:
In the long run I guess. It depends what we do with the "search" extension usage?

Reply:
Once we move that out of core it can use "this" one too, I'm asking to know if more code is needed for this one. We should add a comment that custom fields are not yet migrated to this class (or add them now? I assume the new lib supports them).

Reply:
Almost all code is for export functionality (starting in `createZapHarLog()` and following methods). It looks like it has `setAdditionalField` for custom fields. We would have to use Jackson (or rely on it anyway).

Reply:
Should I be migrating more code or adding a comment here? (Or both?)

Reply:
It's only missing setting the custom fields (iirc).

Reply:
Okay